library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -1340,7 +1341,7 @@ private static DummyTrackOutput createDummyTrackOutput(int id, int type) {
     return new DummyTrackOutput();
   }
 
-  private static final class FormatAdjustingSampleQueue extends SampleQueue {
+  private static final class FormatAdjustingSampleQueue extends HlsCheckedSampleQueue {
```

Comment:
Can we just rename this to `HlsSampleQueue` and have it do both jobs, rather than creating another layer in the class hierarchy?